### DISCUSSION TITLE
Timestamps in Dynamoose

### DISCUSSION THREAD

**Discussion by ghost** [dynamoose/dynamoose#1115] [Q&A]
According to what i understood, on giving timestamps= true, whenever i add a item in the table, 2 attributes createdAt and updatetAt are created for each item that i add in the table.
But in my case only the id,name and location are getting stored and not the timestamps.

```
var dynamoose = require('dynamoose');

module.exports = dynamoose.model('Department',{

    departmentId :{type: String,hashKey : true, required: true},
    departmentName: {type : String,required: true},
    departmentLocation : {type: String, required: true}
},
{
    timestamps: true,
    
});
```
Adding like this

```
  var params = {
            TableName: "Department",
            Item: {
                "departmentId": departmentId,
                "departmentName": departmentName,
                "departmentLocation" : data.departmentLocation
            }
        }
        docClient.put(params, function (err, data) {
            if (err) {
                console.log(err);
                callback({"statuscode":500, "message" : "Error while adding department"})
                return;
            }else{....}
```

**Comment by fishcharlie** [comment]
`timestamps` is a setting for Schema not Model. Your code indicates you are trying to pass it in as a setting on your Model. Move your Schema to a separate variable and set `timestamps` on that Schema, then pass it into your Model.

**Reply by ghost** [reply]
I tried this but the same result. I think its the problem of adding item to the table

```
var schema = new dynamoose.Schema({
    departmentId :{type: String,hashKey : true, required: true},
    departmentName: {type : String,required: true},
    departmentLocation : {type: String, required: true}
},
{
    timestamps: true,
    
})
module.exports =dynamoose.model('Department',schema);
````

**Reply by fishcharlie** [reply]
Ok well there isn't enough information here to debug. So can't really help further. I haven't been able to reproduce this and it works fine for me. Without more clear information really can't do anything else to help.
